modules/addons/deconlog.lua

```diff
@@ -11,79 +11,123 @@ local filepath = "log/decon.log"
 local function add_log(data)
 	game.write_file(filepath, data .. "\n", true, 0) -- write data
 end
-local function get_secs ()
+
+local function get_secs()
 	return format_time(game.tick, { hours = true, minutes = true, seconds = true, string = true })
 end
-local function pos_tostring (pos)
+
+local function pos_to_string(pos)
 	return tostring(pos.x) .. "," .. tostring(pos.y)
 end
 
+local function pos_to_gps_string(pos)
+	return '[gps=' .. tostring(pos.x) .. ',' .. tostring(pos.y) .. ']'
+end
+
+--- Print a message to all players who match the value of admin
+local function print_to_players(admin, message)
+    for _, player in ipairs(game.connected_players) do
+        if player.admin == admin then
+            player.print(message)
+        end
+    end
+end
+
 Event.on_init(function()
 	game.write_file(filepath, "\n", false, 0) -- write data
 end)
 
 if config.decon_area then
 	Event.add(defines.events.on_player_deconstructed_area, function (e)
 		local player = game.get_player(e.player_index)
-		if Roles.player_has_flag(player, "deconlog-bypass") then return end
-		add_log(get_secs() .. "," .. player.name .. ",decon_area," .. pos_tostring(e.area.left_top) .. "," .. pos_tostring(e.area.right_bottom))
+		if Roles.player_has_flag(player, "deconlog-bypass") then
+			return
+		end
+
+		if ((e.area.left_top.x + e.area.right_bottom.x) * (e.area.left_top.y + e.area.right_bottom.y)) > 10000 then
```

Comment:
This maths is wrong. It also would be better to check the number of entites in an area.
```suggestion
		if e.surface.find_entities_filtered{ area = e.area, force = e.player.force, invert = true } > 250 then
```